### PR TITLE
Test coverage settings for consistent code coverage

### PR DESCRIPTION
**Pull request by alistair-singh**
Resolves: SNO-514

### CODE REVIEW

**Review by doubledup — APPROVED** [review]

File: codecov.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+codecov:
+  require_ci_to_pass: yes
+  notify:
+    wait_for_ci: yes
```
[the remark this answers is not in the document]
Comment (yrong) [reply codecov.yml:4]: Yes, seems this is not the critical change.  @alistair-singh Correct me if I'm wrong, I would assume the purpose of this PR seems to ensure always run coverage test for both rust&solidity codes together.

But IMO these two kinds of tests should be isolated and change in solidity code should not trigger a rust test.

Just check codecov `flags` to group coverage reports maybe a better way to handle.
https://docs.codecov.com/docs/flags

[the remark this answers is not in the document]
Comment (alistair-singh) [reply codecov.yml:4]: @doubledup It seems the defaults are broken, the addition of this fixed it for me. I was following this issue: https://community.codecov.com/t/github-status-notification-isnt-delayed-should-wait-all-ci-jobs-to-finish/3708

@yrong Thanks, I actually think flags with a combination of waiting for ci to finish before notify will work. Will try this.

File: .github/workflows/parachain.yml [hunk 2]
```diff
@@ -3,14 +3,14 @@ name: parachain
 on:
   push:
     paths:
-      - "parachain/**"
+      - "*"
       - "!parachain/README.md"
       - "!parachain/LICENSE"
```
[the remark this answers is not in the document]
Comment (alistair-singh) [reply .github/workflows/parachain.yml:8]: I have added the filters back in.

File: .github/workflows/parachain.yml [hunk 3]
```diff
@@ -3,14 +3,14 @@ name: parachain
...
     branches:
       - main
   pull_request:
     paths:
-      - "parachain/**"
+      - "*"
       - "!parachain/README.md"
       - "!parachain/LICENSE"
```
[the remark this answers is not in the document]
Comment (alistair-singh) [reply .github/workflows/parachain.yml:15]: I have added the filters back in.

**Review by yrong — APPROVED** [review]
Awesome!

**Pull request closed without merging**